doc/dune-files.rst

```diff
@@ -848,7 +848,7 @@ files for executables. See `executables_implicit_empty_intf`_.
   available since the 3.0 version of the Dune language.
 
 Linking Modes
-=============
+-------------
```

Comment:
I'm not sure this is right either because then `Linking Modes` will be at the same level as `executable` but it should be its subsection, right?

Perhaps, we should use `~~~~`? See `Handling of the .exe Extension on Windows`, which is a subsection of `install` for an example.

Reply:
(By the way, I now think that maybe it wasn't your PR that broke this... sorry for blaming you wrongly!)

Reply:
Oy. Alrighty maybe this time it'll be right.